### ISSUE TITLE
Add dynamic support

### ISSUE THREAD

**Issue by pjvds** [JoshClose/CsvHelper#43]
For quick-N-dirty CSV reading a dynamic object would be sufficient. This would help people reading a csv file in 3 lines of code:

```
var reader = new CsvReader("mycsv.csv");
var records = reader.GetRecords<dynamic>();
var firstRecord = records.First();

// Get value via name
var foo = firstRecord.ColumnX;

// Get via index (column order)
var bar = firstRecord[0];
```

**Comment by hovermind** [comment]
@JoshClose I want to make my application data agnostic & I tried

var records = csvReader.GetRecords<dynamic>();
foreach (var row in records) {
      log(row.Address);  // error, ExpandoObject does not have Property "Address"
 }
But I saw the _dataArray has Address at intex 3

My app reads csv files written in Japanese. The following works fine:
while(csvReader.Read()){
 var profile = csvReader.GetRecord<CompanyProfile>();
 var address = profile.Address;
}

But the app works for only specific type of data that conforms CompanyProfile class.
I want to make my app data agnostic, if I use csvReader.GetRecords<dynamic>(); then how to get address from csv file?

var records = csvReader.GetRecords<dynamic>();
foreach (var row in records) {
      var address =  row.Adress or row[index] will be fine
 }

How to do it?
